narwhals/utils.py

```diff
@@ -85,17 +85,6 @@ def isinstance_or_issubclass(obj: Any, cls: Any) -> bool:
     return isinstance(obj, cls) or issubclass(obj, cls)
 
 
-def validate_same_library(items: Iterable[Any]) -> None:
-    if all(item._is_polars for item in items):
-        return
-    if all(hasattr(item._compliant_frame, "_implementation") for item in items) and (
-        len({item._compliant_frame._implementation for item in items}) == 1
-    ):
```

Comment:
this function's good, it's just going to be intractable to enforce the "same-library-constraint" everywhere...rather than rewrite it, I'd prefer to just not bother

trying to compare different libraries raises an error anyway, it doesn't do something dangerous like silently give wrong results